include/arc_utilities/dijkstras.hpp

```diff
@@ -735,24 +791,19 @@ namespace arc_dijkstras
                 const AstarNode top_node = queue.top();
                 queue.pop();
                 // Check if the node has already been discovered
-                const auto explored_itr = explored.find(top_node.Index());
-                // We have already been here and it was cheaper then
-                if ((explored_itr != explored.end()) && (top_node.CostToCome() >= explored_itr->second.second))
-                {
-                    continue;
-                }
+                const auto explored_itr = explored.find(top_node.GraphIndex());
                 // We have not been here before, or it is cheaper now
-                else
+                if (!((explored_itr != explored.end()) && (top_node.CostToCome() >= explored_itr->second.second)))
```

Comment:
I don't know if you can. I don't really like the current form either, since order of evaluation isn't defined there's nothing that means the compiler has to evaluate `explored_itr != explored.end()` before `top_node.CostToCome() >= explored_itr->second.second` and if they were evaluated in reverse order that would be undefined behavior. I think the whole thing needs to be rewritten to ensure safety.